Code/Tools/SerializeContextTools/Utilities.cpp

```diff
@@ -191,16 +191,21 @@ namespace AZ::SerializeContextTools
         return result;
     }
 
-    AZStd::string Utilities::GenerateRelativePosixPath(const AZStd::string& projectPath, const AZStd::string& absolutePath)
+    AZStd::string Utilities::GenerateRelativePosixPath(const AZ::IO::PathView& projectPath, const AZ::IO::PathView& absolutePath)
     {
-        AZStd::string absolutePathCpy = absolutePath;
-        AZStd::replace(absolutePathCpy.begin(), absolutePathCpy.end(), AZ::IO::WindowsPathSeparator, AZ::IO::PosixPathSeparator);
-        AZStd::to_lower(absolutePathCpy.begin(), absolutePathCpy.end());
-        return absolutePath.starts_with(projectPath) ? absolutePathCpy.substr(projectPath.length() + 1)
-                                                     : GetStringAfterFirstOccurenceOf("assets/", absolutePathCpy);
+        AZ::IO::FixedMaxPath projectRelativeFilePath = absolutePath.LexicallyProximate(projectPath);
+        AZStd::to_lower(projectRelativeFilePath.Native().begin(), projectRelativeFilePath.Native().end());
+
+        AZStd::string result = projectRelativeFilePath.StringAsPosix();
+        const bool isOutsideProjectFolder = result.starts_with("..");
```

Comment:
Oh, you can use the [AZ::IO::Path::IsRelativeTo](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/IO/Path/Path.h#L269-L270) function to check if the absolutePath is relative to the ProjectPath.

That beats detecting whether the path starts with ".." after getting the relative path.
i.e
```c++
const bool isOutsideProjectFolder = absolutePath.IsRelativeTo(projectPath);
```

Reply:
I tried to use it with the absolute path or on projectRelativeFilePath it does not give me what need :/ (in the capture below we need to enter the if statement)

![image](https://github.com/o3de/o3de/assets/19243508/fc604c25-9119-42c8-b1b0-a826d25a3719)